RevenueCatUI/Templates/V2/ViewHelpers/ShadowModifier.swift

```diff
@@ -49,9 +49,122 @@ struct ShadowModifier: ViewModifier {
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
 extension View {
     func shadow(
-        shadow: ShadowModifier.ShadowInfo
+        shadow: ShadowModifier.ShadowInfo,
+        shape: some Shape
     ) -> some View {
-        self.modifier(ShadowModifier(shadow: shadow))
+        self.modifier(LayerShadowModifier(
+            shape: shape,
+            color: shadow.color,
+            xOffset: shadow.x,
+            yOffset: shadow.y,
+            blur: shadow.radius * 2,
+            spread: 0
+        ))
+    }
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct LayerShadowModifier: ViewModifier {
+    let shape: any Shape
+    let color: Color
+    let xOffset: CGFloat
+    let yOffset: CGFloat
+    let blur: CGFloat
+    let spread: CGFloat
+
+    func body(content: Content) -> some View {
+        content
+            .background {
+                GeometryReader { geometry in
+                    let rect = geometry.frame(in: .local)
+                    LayerShadowView(shape: shape,
+                                    color: color,
+                                    xOffset: xOffset,
+                                    yOffset: yOffset,
+                                    blur: blur,
+                                    spread: spread,
+                                    rect: rect)
+                    .frame(maxWidth: .infinity, maxHeight: .infinity)
+                }
+            }
+    }
+}
+
+// Using the .shadow() modified to add a drop shadow in SwiftUI has multiple downsides:
+// - The shadow is applied to all children views (can be worked around with .compositingGroup())
+// - The `.compositingGroup()` workaround stops working if the view has less than 100% opacity.
+// - The shadow inherits the opacity of the view, making it impossible to have a translucent
+//   or transparent view with a 100% opacity shadow.
+//
+// ShadowUIView tries to work around these limitations by rendering the shadow in a backing UIView,
+// and using a Shape to mask off the inner part of the shadow.
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+private struct LayerShadowView: UIViewRepresentable {
```

Comment:
This is the main addition of the PR and the way we fix SwiftUI shadows by using a CALayer shadow instead. Documented in code as it feels like important context.